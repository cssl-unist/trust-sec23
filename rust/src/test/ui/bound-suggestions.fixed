// run-rustfix

#[allow(unused)]
use std::fmt::Debug;
// Rustfix should add this, or use `std::fmt::Debug` instead.

#[allow(dead_code)]
fn test_impl(t: impl Sized + Debug) {
    println!("{:?}", t);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_no_bounds<T: Debug>(t: T) {
    println!("{:?}", t);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_one_bound<T: Sized + Debug>(t: T) {
    println!("{:?}", t);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_no_bounds_where<X, Y>(x: X, y: Y) where X: std::fmt::Debug, Y: Debug {
    println!("{:?} {:?}", x, y);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_one_bound_where<X>(x: X) where X: Sized + Debug {
    println!("{:?}", x);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_many_bounds_where<X>(x: X) where X: Sized, X: Sized, X: Debug {
    println!("{:?}", x);
    //~^ ERROR doesn't implement
}

pub fn main() { }
