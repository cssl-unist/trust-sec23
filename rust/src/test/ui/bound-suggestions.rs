// run-rustfix

#[allow(unused)]
use std::fmt::Debug;
// Rustfix should add this, or use `std::fmt::Debug` instead.

#[allow(dead_code)]
fn test_impl(t: impl Sized) {
    println!("{:?}", t);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_no_bounds<T>(t: T) {
    println!("{:?}", t);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_one_bound<T: Sized>(t: T) {
    println!("{:?}", t);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_no_bounds_where<X, Y>(x: X, y: Y) where X: std::fmt::Debug, {
    println!("{:?} {:?}", x, y);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_one_bound_where<X>(x: X) where X: Sized {
    println!("{:?}", x);
    //~^ ERROR doesn't implement
}

#[allow(dead_code)]
fn test_many_bounds_where<X>(x: X) where X: Sized, X: Sized {
    println!("{:?}", x);
    //~^ ERROR doesn't implement
}

pub fn main() { }
