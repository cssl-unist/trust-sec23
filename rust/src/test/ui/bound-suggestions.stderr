error[E0277]: `impl Sized` doesn't implement `Debug`
  --> $DIR/bound-suggestions.rs:9:22
   |
LL |     println!("{:?}", t);
   |                      ^ `impl Sized` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = note: required by `std::fmt::Debug::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
   |
LL | fn test_impl(t: impl Sized + Debug) {
   |                            ^^^^^^^

error[E0277]: `T` doesn't implement `Debug`
  --> $DIR/bound-suggestions.rs:15:22
   |
LL |     println!("{:?}", t);
   |                      ^ `T` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = note: required by `std::fmt::Debug::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
   |
LL | fn test_no_bounds<T: Debug>(t: T) {
   |                    ^^^^^^^

error[E0277]: `T` doesn't implement `Debug`
  --> $DIR/bound-suggestions.rs:21:22
   |
LL |     println!("{:?}", t);
   |                      ^ `T` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = note: required by `std::fmt::Debug::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
   |
LL | fn test_one_bound<T: Sized + Debug>(t: T) {
   |                            ^^^^^^^

error[E0277]: `Y` doesn't implement `Debug`
  --> $DIR/bound-suggestions.rs:27:30
   |
LL |     println!("{:?} {:?}", x, y);
   |                              ^ `Y` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = note: required by `std::fmt::Debug::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting type parameter `Y`
   |
LL | fn test_no_bounds_where<X, Y>(x: X, y: Y) where X: std::fmt::Debug, Y: Debug {
   |                                                                   ^^^^^^^^^^

error[E0277]: `X` doesn't implement `Debug`
  --> $DIR/bound-suggestions.rs:33:22
   |
LL |     println!("{:?}", x);
   |                      ^ `X` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = note: required by `std::fmt::Debug::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting this bound
   |
LL | fn test_one_bound_where<X>(x: X) where X: Sized + Debug {
   |                                                 ^^^^^^^

error[E0277]: `X` doesn't implement `Debug`
  --> $DIR/bound-suggestions.rs:39:22
   |
LL |     println!("{:?}", x);
   |                      ^ `X` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = note: required by `std::fmt::Debug::fmt`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider further restricting type parameter `X`
   |
LL | fn test_many_bounds_where<X>(x: X) where X: Sized, X: Sized, X: Debug {
   |                                                            ^^^^^^^^^^

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0277`.
